examples/mxnet/gcmc/utils.py

```diff
@@ -0,0 +1,183 @@
+    print("All Logs will be saved to %s" %logpath)
+    logging.root.setLevel(level)
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    logfile = logging.FileHandler(logpath)
+    logfile.setLevel(level)
+    logfile.setFormatter(formatter)
+    logging.root.addHandler(logfile)
+    if not no_console:
+        # Initialze the console logging
+        logconsole = logging.StreamHandler()
+        logconsole.setLevel(console_level)
+        logconsole.setFormatter(formatter)
+        logging.root.addHandler(logconsole)
+    return folder
+
+
+def parse_ctx(ctx_args):
+    ctx = re.findall('([a-z]+)(\d*)', ctx_args)
+    ctx = [(device, int(num)) if len(num) > 0 else (device, 0) for device, num in ctx]
+    ctx = [mx.Context(*ele) for ele in ctx]
+    return ctx
+
+
+def gluon_total_param_num(net):
+    return sum([np.prod(v.shape) for v in net.collect_params().values()])
+
+
+def gluon_net_info(net, save_path=None):
+    info_str = 'Total Param Number: {}\n'.format(gluon_total_param_num(net)) +\
+               'Params:\n'
+    for k, v in net.collect_params().items():
+        info_str += '\t{}: {}, {}\n'.format(k, v.shape, np.prod(v.shape))
+    info_str += str(net)
+    if save_path is not None:
+        with open(save_path, 'w') as f:
+            f.write(info_str)
+    return info_str
+
+
+def params_clip_global_norm(param_dict, clip, ctx):
+    grads = [p.grad(ctx) for p in param_dict.values()]
+    gnorm = gluon.utils.clip_global_norm(grads, clip)
+    return gnorm
+
+
+def get_global_norm(arrays):
+    ctx = arrays[0].context
+    total_norm = nd.add_n(*[nd.dot(x, x).as_in_context(ctx)
+                            for x in (arr.reshape((-1,)) for arr in arrays)])
+    total_norm = nd.sqrt(total_norm).asscalar()
+    return total_norm
+
+
+def div_up(a, b):
+    return (a + b - 1) // b
+
+
+def copy_to_ctx(data, ctx, dtype=None):
+    if isinstance(data, (list, tuple)):
+        if dtype is None:
+            dtype = data[0].dtype
+        return [nd.array(ele, dtype=dtype, ctx=ctx) for ele in data]
+    elif isinstance(data, dict):
+        if dtype is None:
+            return {k: copy_to_ctx(v, ctx) for k, v in data.items()}
+        else:
+            return {k: copy_to_ctx(v, ctx, dtype) for k, v in data.items()}
+    else:
+        if dtype is None:
+            dtype = data.dtype
+        return nd.array(data, dtype=dtype, ctx=ctx)
+
+
+def get_activation(act):
+    """Get the activation based on the act string
+
+    Parameters
+    ----------
+    act: str or HybridBlock
+
+    Returns
+    -------
+    ret: HybridBlock
+    """
+    if act is None:
+        return lambda x: x
+    if isinstance(act, str):
+        if act == 'leaky':
+            return nn.LeakyReLU(0.1)
+        elif act == 'identity':
+            return IdentityActivation()
+        elif act == 'elu':
+            return ELU()
+        elif act in ['relu', 'sigmoid', 'tanh', 'softrelu', 'softsign']:
```

Comment:
`IdentityActivation` and `ELU` are not declared or imported anywhere.

Reply:
Done